src/System.Configuration.ConfigurationManager/src/System/Configuration/GenericEnumConverter.cs

```diff
@@ -32,35 +33,54 @@ public override object ConvertFrom(ITypeDescriptorContext ctx, CultureInfo ci, o
             try
             {
                 string value = (string)data;
-                if (string.IsNullOrEmpty(value)) throw new Exception();
+                bool valueIsInvalid = string.IsNullOrEmpty(value);
 
                 // Disallow numeric values for enums.
-                if (!string.IsNullOrEmpty(value) &&
-                    (char.IsDigit(value[0]) ||
-                    (value[0] == '-') ||
-                    (value[0] == '+')))
-                    throw new Exception();
+                if (!valueIsInvalid &&
+                    (char.IsDigit(value[0]) || (value[0] == '-') || (value[0] == '+')))
+                {
+                    valueIsInvalid = true;
+                }
 
-                if (value != value.Trim())
+                if (!valueIsInvalid)
```

Comment:
Again, I would have done `throw CreateExceptionForInvalidValue()` above. If there's a reason for not doing that then this line could just be `else`